### ISSUE TITLE
[Proposal] Support port mapping in tor/listen.go

### ISSUE THREAD

**Issue by parazyd** [cretz/bine#30]
This is a feature request.

Would it be possible to support port maps like stem does in `tor/listen.go`?

For example, in stem we can do:
```
service = controller.create_ephemeral_hidden_service({80: 5000, 90:5001}, await_publication = True)
```

This would have two listeners, on ports 5000 and 5001, and it would map them to 80 and 90.

**Comment by cretz** [comment]
Yeah, today I only support one `LocalPort`/`LocalListener` in the simplified listen API in the `tor` package (but we do support multiple remote ports).  There's no specific reason not to accept more ports/listeners (e.g. `AdditionalLocalPorts`/`AdditionaLocalListeners` to retain backwards compatibility). I haven't been able to work on this project in a while, so who knows if/when it'll be impl'd. In the meantime, code can use the lower-level pieces to communicate directly w/ the control port using the `control` package the way the `Listen` call does.

**Comment by cretz** [comment]
if I revisit this, I think I may accept a port-mappings structure that will be mutually exclusive with the existing local/remote port pieces. This will give the most flexibility.
